test/610_proxy_wait_for_weave_test.sh

```diff
@@ -0,0 +1,17 @@
+#! /bin/bash
+
+. ./config.sh
+
+start_suite "Proxy waits for weave to be ready before running container commands"
+weave_on $HOST1 launch-proxy
+BASE_IMAGE=ubuntu
+# Ensure the base image does not exist, so that it will be pulled
+if (docker_on $HOST1 images $BASE_IMAGE | grep $BASE_IMAGE); then
+  docker_on $HOST1 rmi $BASE_IMAGE
+fi
+
+assert_raises "docker_proxy_on $HOST1 run -e 'WEAVE_CIDR=10.2.1.1/24' --name=c1 $BASE_IMAGE ip link show ethwe" 0
+assert "docker_on $HOST1 inspect --format='{{.Config.Entrypoint}}' c1" "[/home/weavewait/weavewait]"
+assert "docker_on $HOST1 inspect --format='{{.Config.Cmd}}' c1" "[ip link show ethwe]"
+
+end_suite
```

Comment:
This whole test is too low-level. Smoke tests are meant to simulate what a user/app would do. So I think the first line is enough, though of course it could succeed by luck of timing. Nothing we can do about w/o jumping through hoops.